spec/requests/user_posts_spec.rb

```diff
@@ -1,7 +1,33 @@
 require 'rails_helper'
 
-RSpec.describe 'UserPosts', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+RSpec.describe UserPostsController, type: :request do
+  describe 'GET #index' do
+    it 'returns a successful response' do
+      get user_user_posts_path(user_id: 1)
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a successful response' do
+      get user_user_post_path(user_id: 1, id: 1)
+      expect(response).to have_http_status(:success)
+    end
+  end
+end
+
+RSpec.describe UsersController, type: :request do
+  describe 'GET #index' do
+    it 'returns a successful response' do
+      get users_path
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a successful response' do
+      get user_path(id: 1)
+      expect(response).to have_http_status(:success)
+    end
   end
 end
```

Comment:
- [ ] It's a requirement that you include tests that ensure the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

Kindly do this for the index and show views for the Post as well